Replicating figure 5

eotamm opened the pull request.

eotamm committed:
Add files via upload

eotamm committed:
Update README.md

antagomir commented on Salehi2025/Figure5.qmd:
```diff
@@ -0,0 +1,450 @@
+editor: visual
+---
+
+# Indroduction
```
Indro -> Intro?

antagomir commented on Salehi2025/Figure5.qmd:
````diff
@@ -0,0 +1,450 @@
+Now we filter out samples with missing gender or age information, extracts relevant metadata, computes total read counts per sample, and applies a log-transformation to the ARG load variable. These steps prepare the dataset for downstream analysis and modeling.
+
+```{r, Readcount}
+# Filter samples with complete gender and age data
+non_na_samples <- !is.na(colData(tse)$gender) & !is.na(colData(tse)$host_age_years)
+TSE_filtered <- tse[, non_na_samples]
+
+# Extract metadata
+adult_metadata <- as.data.frame(colData(TSE_filtered))
+
+# Readcount
+counts_mat <- assay(TSE_filtered, "counts")
+adult_metadata$readcount_2 <- colSums(counts_mat)
+
+adult_metadata$log_ARG_load <- log(adult_metadata$ARG_load)
+```
+
````
Is this something that should be done already before fitting the model in Figure5models.R?

In general, separate the data processing script from the final report that could assume readily processed data.

You could have for isntance:

data.R -> data preparation
model.R -> model fitting
report.qmd -> reporting with explanations, figures, tables

antagomir commented on Salehi2025/Figure5.qmd:
```diff
@@ -0,0 +1,450 @@
+counts_mat <- assay(TSE_filtered, "counts")
+adult_metadata$readcount_2 <- colSums(counts_mat)
+
+adult_metadata$log_ARG_load <- log(adult_metadata$ARG_load)
```
check if we really need the separate variable for log_ARG_load or can we deal with just ARG_load (and call log-normal models etc as needed)

antagomir commented on Salehi2025/Figure5.qmd:
````diff
@@ -0,0 +1,450 @@
+```{r, AB classes}
+# Extract count matrix and metadata
+counts_mat <- assay(tse, "counts")
+row_df <- as.data.frame(rowData(tse))  # Gene annotations
+col_df <- as.data.frame(colData(tse)) %>% mutate(sample = acc)  # Sample metadata
+
+# Set sample names as column names
+colnames(counts_mat) <- col_df$acc 
+
+# Merge gene-level counts with their corresponding antibiotic resistance class
+counts_df <- as.data.frame(counts_mat) %>%
+  mutate(gene = rownames(counts_mat)) %>%
+  left_join(row_df[, c("GENE", "Class")], by = c("gene" = "GENE")) %>%
+  pivot_longer(cols = -c(gene, Class), names_to = "sample", values_to = "count")
+
+# Aggregate counts at sample and antibiotic class level
+agg_counts <- counts_df %>%
+  group_by(sample, Class) %>%
+  summarize(class_total = sum(count, na.rm = TRUE), .groups = "drop")
+
+# Keep only the top 5 most frequent antibiotic classes
+top_5_AB_classes <- c(
+  "Tetracycline", 
+  "Beta-lactam", 
+  "Macrolide, Lincosamide, Streptogramin B", 
+  "Aminoglycoside", 
+  "Amphenicol"
+)
+
+agg_counts <- agg_counts %>% filter(Class %in% top_5_AB_classes)
````
same comment, prepare data in a separate file before the report.

antagomir commented on Salehi2025/Figure5.qmd:
```diff
@@ -0,0 +1,450 @@
+# Subset metadata to match samples with non-zero data
+adult_metadata <- colData(TSE_adult_filtered)[non_empty_samples, ]
+
+# Calculate Shannon diversity for each sample (based on species richness and evenness)
```
Shannon is alpha diversity, not beta diversity like the title says?

antagomir commented on Salehi2025/Figure5.qmd:
````diff
@@ -0,0 +1,450 @@
+This section prepares the data for beta diversity analysis and computes Shannon diversity index from relative abundance profiles.
+
+```{r, Beta Diversity}
+# Filter metadata to include only samples with GDP and antibiotic usage information
+adult_metadata <- adult_metadata %>%
+  mutate(
+    continent = factor(geo_loc_name_country_continent_calc)
+  ) %>%
+  filter(!is.na(Usage), !is.na(GDP_per_head))
+
+# Keep only samples with non-missing GDP and antibiotic usage in the TSE object
+non_na_conditions <- !is.na(colData(TSE_filtered)$GDP_per_head) &
+                     !is.na(colData(TSE_filtered)$Usage)
+TSE_adult_filtered <- TSE_filtered[, non_na_conditions]
+
+# Update the sample metadata in the filtered TSE object
+colData(TSE_adult_filtered) <- DataFrame(adult_metadata)
+
+# Extract relative abundance data
+assay_data_clean <- assay(TSE_adult_filtered, "relabundance")
+
+# Remove samples that contain only zero abundances
+non_empty_samples <- colSums(assay_data_clean > 0, na.rm = TRUE) > 0
+assay_data_clean <- assay_data_clean[, non_empty_samples]
+
+# Subset metadata to match samples with non-zero data
+adult_metadata <- colData(TSE_adult_filtered)[non_empty_samples, ]
````
move all data preparation into separate script and see if there is something that can be shortedned / removed / cleaned up

antagomir commented on Salehi2025/Figure5.qmd:
````diff
@@ -0,0 +1,450 @@
+```{r, helpers}
+## Helpers
+
+# Format numbers: Convert values below 0.001 to "<0.001"
+to_neat <- function(x, threshold = 0.001) {
+  ifelse(x > threshold, as.character(x), "<0.001")
+}
+
+# Assign significance stars for confidence intervals (log-scale)
+get_stars_CI_log <- function(lower, upper) {
+  ifelse(lower > 1 | upper < 1, "*", "ns")
+}
+
+# Assign significance stars for confidence intervals (linear scale)
+get_stars_CI <- function(lower, upper) {
+  ifelse(lower > 0 | upper < 0, "*", "ns")
+}
+```
````
utility functions could also go to their own file, e.g. `funcs.R`

antagomir commented on Salehi2025/Figure5.qmd:
````diff
@@ -0,0 +1,450 @@
+}
+```
+
+# Modeling
+
+First, we define the response variables (ARG_load, shannon_diversity) and income groups (LMIC, HIC) used for stratified statistical modeling.
+
+```{r}
+responses <- c("ARG_load", "shannon_diversity")
+incomes <- 0:1
+set.seed(123123)
+```
+
````
Oveall seems good & clear explanations.

antagomir commented on Salehi2025/Figure5.qmd:
````diff
@@ -0,0 +1,450 @@
+The resulting model objects are saved as `.rds` files (e.g. `fit1.rds`, `fit2.rds`, etc.) and loaded back into the this Quarto document for result summarization and plotting.
+
+```{r, models}
+# Load models
+fit_list1 <- readRDS("fit1.rds")
+fit_list2 <- readRDS("fit2.rds")
+fit_list3 <- readRDS("fit3.rds")
+fit_list4 <- readRDS("fit4.rds")
+
````
if possible, put it in its own chunk (not sure if possible?)

antagomir replied:
some more explanation what each different model / summary below does would help the reader to follow

antagomir commented on Salehi2025/Figure5models.qmd:
```diff
@@ -0,0 +1,304 @@
+---
+title: "Figure5 Models"
+format: html
+editor: visual
```
Could be just R file

antagomir reviewed (commented):
Thanks!

Some notes.

1) README could also tell what command to use to run the documents. To speed up testing.

2) Model fitting (Figure5models.qmd) can also be plain R for easier testing unless it has some particular content that prefers qmd format. In this case it does not seem to have.

3) I did not manage to replicate those runs: 
```
knitr::purl("Figure5models.qmd")
source("Figure5models.R")

```Error in eval(ei, envir) : object 'responses' not found

Some other comments added.

eotamm committed:
Delete Salehi2025/Figure5.qmd

eotamm committed:
Delete Salehi2025/Figure5models.qmd

eotamm committed:
Add files via upload

eotamm committed:
Update README.md

eotamm committed:
Update report.qmd

eotamm commented:
I updated the README and report.qmd with clearer instructions and structure. Added new scripts: data.R, funcs.R, and model.R. All steps can now be followed directly from the report.qmd.

eotamm committed:
Update README.md

eotamm committed:
Delete Salehi2025/report.qmd

eotamm committed:
Add files via upload

eotamm commented:
I made the changes and tested running source('main.R') and it worked.

antagomir commented:
Great, we can merge this and let's discuss possible later improvents separately.

antagomir commented:
Did you resolve all the suggestions ab0ve?

eotamm commented:
Yeah all of those should be resolved.

antagomir commented:
yess. the usual convention is to click each "resolve" button to confirm they have been resolved

eotamm commented:
Okay, good to know in the future.
